src/components/PinInput/src/PinInput.vue

```diff
@@ -0,0 +1,293 @@
+		},
+
+		getPinCellRef(index) {
+			return `pin-cell-${index}`;
+		},
+
+		// Focus on verification code cell input at given index.
+		focusOnPinCell(index) {
+			const cellReference = this.$refs[this.getPinCellRef(index)];
+			if (cellReference?.[0]?.focus) {
+				cellReference[0].focus();
+			}
+		},
+
+		onInputPin(event, index) {
+			if (!event.data) {
+				return;
+			}
+
+			// Only allow integers in input
+			const BASE_TEN = 10;
+			const inputValue = Number.isInteger(Number.parseInt(event.data, BASE_TEN)) ? event.data : '';
+
+			// One-time-code autofill is treated as an input, not a paste
+			if (this.attemptSplitPinIntoInputs(inputValue, index)) {
+				return;
+			}
+
+			this.$set(this.pin, index, inputValue);
+
+			const firstIncompleteCellIndex = this.findFirstIncompletePinCellIndex();
+
+			// eslint-disable-next-line no-magic-numbers
+			if (firstIncompleteCellIndex === -1) {
+				this.handleComplete(this.currentPin);
+				return;
+			}
+
+			// This allows us to auto jump to the next code cell input as the user types.
+			this.focusOnPinCell(firstIncompleteCellIndex);
+		},
+
+		attemptSplitPinIntoInputs(value, inputIndex) {
+			const BASE_TEN = 10;
+			if (inputIndex === 0
+				&& value?.length === this.pinLength
+				&& Number.isInteger(Number.parseInt(value, BASE_TEN))
+			) {
+				this.$set(this, 'pin', value.split(''));
+
+				// Having a timeout here gives the user a chance to see their code before success/failure
+				const TIMEOUT_LENGTH_MS = 500;
+				setTimeout(() => { this.handleComplete(this.currentPin); }, TIMEOUT_LENGTH_MS);
+				return true;
+			}
+			return false;
+		},
+
+		handleComplete(pin) {
+			this.$emit('complete', pin);
+		},
+
+		onPastePin(event, index) {
+			event.preventDefault();
+			const pastedValue = (event.clipboardData || window.clipboardData).getData('text');
+			this.attemptSplitPinIntoInputs(pastedValue, index);
+		},
+
+		onDelete(event, index) {
+			// Reset current index, and then focus on previous cell
+			this.$set(this.pin, index, '');
+
+			// eslint-disable-next-line no-magic-numbers
+			const newIndex = index - 1;
+			if (newIndex >= 0) {
+				this.focusOnPinCell(newIndex);
+			}
+		},
+
+		onFocusPin(event, index) {
+			event.preventDefault();
+			this.$set(this.pin, index, '');
+		},
+
+		resetPin() {
+			const newArray = (new Array(this.pinLength)).fill('');
+			this.$set(this, 'pin', newArray);
+		},
+
+		shakeAndClearInputs() {
+			this.resetPin();
+			this.focusOnPinCell(0);
+			this.isShaking = true;
+			const TIMEOUT_LENGTH_MS = 1000;
+			setTimeout(() => {
+				this.isShaking = false;
+			}, TIMEOUT_LENGTH_MS);
+		},
+	},
+};
+</script>
+
+<style module="$s">
+.PinInputContainer {
+	display: flex;
+	flex-wrap: nowrap;
+	align-items: center;
+	justify-content: space-between;
+
+	&.error {
+		padding-bottom: 8px;
+	}
+}
+
+.PinInput {
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	box-sizing: border-box;
+	width: 50px;
+	height: 50px;
```

Comment:
just to double-check, the width and height are suppose to be `50px` and not `48px` right?

Reply:
yeah, it was 46px + 2px potential border on each side, solved using the box-sizing you suggested. Ive never used that before so that was good to know